### ISSUE TITLE
Mixer fails to startup without Galley

### ISSUE THREAD

**Issue by howardjohn** [istio/istio#14841]
Installed with `--set global.useMCP=false --set galley.enabled=false`

Mixer policy and telemetry crash with logs

`2019-06-14T14:34:50.570279Z     error   istio.io/istio/mixer/pkg/config/crd/store.go:145: Failed to list *unstructured.Unstructured: the server could not find the requested resource`

Version: master-20190614-09-15

**Comment by kyessenov** [comment]
cc @ozevren @douglas-reid 
I believe the plan was to remove the direct k8s config store from Mixer in 1.2. 
We no longer test it AFAICT.

**Comment by howardjohn** [comment]
Related issue https://github.com/istio/istio/issues/14836.

**Comment by douglas-reid** [comment]
FWIW, this was on the proposed roadmap for things to be dealt with in 1.2, but was never prioritized.

**Comment by howardjohn** [comment]
Is fixing this wasted effort then? Will Galley be required in 1.3?

**Comment by ozevren** [comment]
That's the original plan. Besides, Mixer v2 plans make the current Mixer implementation obsolete. Unless there is a customer blocking issue, I'd suggest keeping its priority lower and punting it.

**Comment by howardjohn** [comment]
My question was (in my head only) more about #14836. It seems pilot doesn't do validation without galley but a lot of pilot code expects valid config. Is it worth messing with the direct kube crd reader to make it so validation?

**Comment by ozevren** [comment]
The plan is to remove the CRD code from Pilot and make Galley required. I don't think it is worth plumbing the validation in the (soon to be) old stack.

**Comment by Stono** [comment]
Hey, 
So I've just hit this.  I'm worried about comments like "We no longer test it AFAICT.".  Not running with galley is an option exposed to customers via the installer.  We've never ran with galley as we haven't been able to allocate the time into testing a deployment with it.

  - If it's required, we should remove the option to disable it.
  - If it isn't required, it should absolutely be tested.

**Comment by ozevren** [comment]
Galley wasn't required in 1.2.0. However, we're going in a direction to make it required. This seems like code rot, more than anything else.

**Comment by ozevren** [comment]
This should fix this: https://github.com/istio/istio/pull/15033
@howardjohn Could you verify, if possible?

**Comment by howardjohn** [comment]
@ozevren lgtm
```
2019-06-20T20:47:16.991077Z     warn    Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2019-06-20T20:47:16.991298Z     info    Kinds: []string{"adapter", "template", "instance", "handler", "rule", "attributemanifest"}
2019-06-20T20:47:16.994943Z     info    Awaiting for config store sync...
gc 3 @0.021s 10%: 0.044+4.9+0.005 ms clock, 0.26+5.5/6.0/0.001+0.033 ms cpu, 5->7->5 MB, 6 MB goal, 6 P
2019-06-20T20:47:17.005068Z     info    Starting runtime config watch...
2019-06-20T20:47:17.005199Z     info    Built new config.Snapshot: id='0'
2019-06-20T20:47:17.005212Z     info    Cleaning up handler table, with config ID:-1
gc 4 @0.038s 8%: 0.009+3.7+0.005 ms clock, 0.054+2.1/3.2/7.3+0.034 ms cpu, 7->8->4 MB, 10 MB goal, 6 P
2019-06-20T20:47:17.015499Z     info    Built new config.Snapshot: id='1'
2019-06-20T20:47:17.015546Z     info    adapters        getting kubeconfig from: ""     {"adapter": "kubernetesenv.istio-system"}
2019-06-20T20:47:17.015552Z     warn    Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2019-06-20T20:47:17.016112Z     info    adapters        Waiting for kubernetes cache sync...    {"adapter": "kubernetesenv.istio-system"}
gc 5 @0.047s 8%: 0.007+4.1+0.011 ms clock, 0.043+1.9/3.3/3.3+0.071 ms cpu, 8->10->6 MB, 9 MB goal, 6 P
2019-06-20T20:47:17.116249Z     info    adapters        Cache sync successful.  {"adapter": "kubernetesenv.istio-system"}
2019-06-20T20:47:17.116303Z     info    adapters        getting kubeconfig from: ""     {"adapter": "kubernetesenv.istio-system"}
2019-06-20T20:47:17.116310Z     warn    Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
2019-06-20T20:47:17.117589Z     info    Setting up event handlers
2019-06-20T20:47:17.117710Z     info    Starting Secrets controller
2019-06-20T20:47:17.117746Z     info    Waiting for informer caches to sync
2019-06-20T20:47:17.118835Z     info    adapters        serving prometheus metrics on 42422     {"adapter": "prometheus.istio-system"}
2019-06-20T20:47:17.121301Z     info    Cleaning up handler table, with config ID:0
2019-06-20T20:47:17.121573Z     info    Starting monitor server...
gc 6 @0.577s 1%: 0.009+9.2+0.004 ms clock, 0.059+1.5/9.5/3.1+0.027 ms cpu, 11->11->6 MB, 13 MB goal, 6 P
Istio Mixer: root@5827f33e-9388-11e9-9590-0a580a2c0406-gcr.io/istio-testing-6de4a2b9ffc6cfdd43c4b4e3d53f3ecb187ba1f8-6de4a2b9ffc6cfdd43c4b4e3d53f3ecb187ba1f8-Clean
Starting gRPC server on port 9091
2019-06-20T20:47:22.203511Z     info    ControlZ available at 127.0.0.1:987
```
```
NAME                                                              READY   STATUS      RESTARTS   AGE
grafana-6fb9f8c5c7-tv99z                                          1/1     Running     0          59s
istio-citadel-649554bdb-t4nvt                                     1/1     Running     0          59s
istio-cleanup-secrets-5dcc4961ae2980f452e00e2d9aa183b5-tfzlm      0/1     Completed   0          59s
istio-grafana-post-install-5dcc4961ae2980f452e00e2d9aa183bl747m   0/1     Completed   0          59s
istio-ingressgateway-76469bcfb-r2sgv                              1/1     Running     0          59s
istio-init-crd-10-j5tb2                                           0/1     Completed   0          80s
istio-init-crd-11-9dcj7                                           0/1     Completed   0          80s
istio-init-crd-12-swsrm                                           0/1     Completed   0          80s
istio-pilot-fff4f858d-597rd                                       2/2     Running     0          59s
istio-policy-556f674d4f-vngx2                                     2/2     Running     0          59s
istio-security-post-install-5dcc4961ae2980f452e00e2d9aa183ff6xl   0/1     Completed   0          59s
istio-sidecar-injector-56fb7b56bd-hh9cv                           1/1     Running     0          58s
istio-telemetry-b7548945d-r98w2                                   2/2     Running     0          59s
prometheus-776fdf7479-djp74                                       1/1     Running     0          59s
```
